contrib/pkg/installmanager/installmanager.go

```diff
@@ -206,7 +216,17 @@ func (m *InstallManager) waitForInstallerBinaries() {
 
 // cleanupBeforeInstall allows recovering from an installation error and allows retries
 func (m *InstallManager) cleanupBeforeInstall() error {
-	// first cleanup any remnant cache files from a previous installation attempt
+	err := m.cleanupMetadataConfigmap()
+	if err != nil {
+		return err
+	}
+
+	err = m.cleanupAdminKubeconfigSecret()
+	if err != nil {
+		return err
+	}
```

Comment:
addressed feedback and added tests as a separate commit. kubebuilder's NewFakeClient() doesn't allow the same level of mocking/expectations as the regular client-go, so some refactoring was necessary. curious to hear what you think of the approach.

Reply:
Looks great, it would be cool if we could do the same approach without detaching the functions from the install manager but I don't know if that's possible. Thanks for addressing!

Reply:
since kubebuilder goclient doesn't let us mock/expect, i couldn't come up with another way other than to allow us to easily replace these functions for test purposes. discussion about extending fake client to support this can be found here https://github.com/kubernetes-sigs/controller-runtime/issues/72